### ISSUE TITLE
Ability to export symbols without defining them in zig

### ISSUE THREAD

**Issue by RinLovesYou** [ziglang/zig#18531]
Pretty niche usecase here, but the ability exists in other languages.

Say we define some function in global assembly
```zig
comptime {
    asm (
        \\.global Foo
        \\Foo:
        \\    ;some function logic here
    )
}
```

Compile to a shared library, and see the function not exported.

One solution would be to expose `export_symbol_names` for `*Compile` for other targets besides WASM.
Linking .DEF files would be another option, although i believe the `export_symbol_names` approach would be nicer.

**Comment by mlugg** [comment]
Does this work for your use case?
```zig
export fn Foo() callconv(.Naked) void {
    asm (
        \\ ; function logic here
    );
}
```

**Comment by RinLovesYou** [comment]
> Does this work for your use case?
> 
> ```zig
> export fn Foo() callconv(.Naked) void {
>     asm (
>         \\ ; function logic here
>     );
> }
> ```

Yes, I am aware of that method,that is the alternative here. For the moment i will be writing a source generator that does this, however the option to export symbols defined only in asm would still be nice to have!

**Comment by mlugg** [comment]
Is there a use case for directly exporting symbols defined in asm which *isn't* handled by existing language features? I can't immediately see one, but I could be missing something.

**Comment by Rexicon226** [comment]
Why are you exporting them through zig? Why not link an external asm file/object?
